modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/Filters.tsx

```diff
@@ -95,21 +97,30 @@ function AddFDSFilterModalContent({
 	);
 	const [picklists, setPicklists] = useState<IPickList[]>([]);
 	const [preselectedValues, setPreselectedValues] = useState<any[]>([]);
-	const [saveButtonDisabled, setSaveButtonDisabled] = useState<boolean>();
-	const [selectedField, setSelectedField] = useState<IField | null>(
-		fields.find((item) => item.name === filter?.fieldName) || null
+	const [saveButtonDisabled, setSaveButtonDisabled] = useState<boolean>(
+		filter ? false : true
+	);
+	const [selectedField, setSelectedField] = useState<IField | undefined>(
+		fields.find((item) => item.name === filter?.fieldName)
 	);
 	const [selectedPicklist, setSelectedPicklist] = useState<IPickList>();
+	const [sourceType, setSourceType] = useState<
+		ESelectionFilterSourceType | undefined
+	>();
 	const [to, setTo] = useState<string>((filter as IDateFilter)?.to ?? '');
 
+	const inUseFields: (string | undefined)[] = fields.map((item) =>
+		fieldNames?.includes(item.name) ? item.name : undefined
+	);
+
 	useEffect(() => {
 		getAllPicklists().then((items) => {
 			setPicklists(items);
 
 			const picklist = items.find(
 				(item) =>
 					String(item.externalReferenceCode) ===
-					(filter as any)?.listTypeDefinitionERC
+					(filter as any)?.source
```

Comment:
Dani noticed this in an unfortunately deleted comment, this should be covered by FF.

Reply:
yep I deleted it becasue I wasn't 100% sure then I talked to Miguel and rechecked. He's going to add the FF check here and when saving